src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
+Animal.removeFromAlive = function(target) {
+  const index = this.alive.indexOf(target);
+
+  if (index !== -1) {
+    this.alive.splice(index, 1);
+  }
+};
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  decreaseHealth(amount) {
+    this.health -= amount;
+
+    if (this.health <= 0) {
+      Animal.removeFromAlive(this);
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(target) {
+    if (target instanceof Herbivore && !target.hidden) {
+      target.decreaseHealth(50);
```

Comment:
Is it necessary to create decreaseHealth function? You use it only once. 
Also you can make const biteValue = 50 and use it insted of premitives